packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -2274,6 +2303,13 @@ class TimePickerDialog extends StatefulWidget {
   /// {@macro flutter.material.time_picker.switchToTimerEntryModeIcon}
   final Icon? switchToTimerEntryModeIcon;
 
+  /// If true and entry mode is [TimePickerEntryMode.input], the hour and minute
+  /// fields will be empty on start instead of pre-filled with [initialTime].
+  ///
+  /// Improves UX by removing the need to delete default values before typing.
+  /// Skipped in dial mode.
```

Comment:
```suggestion
  /// Has no effect in dial mode.
```
Nit: Generally we try and avoid putting opinions into the documentation like, "do this for good UX".